classes/06class/exercises/c06-serverless01/alanlima/output.md

```diff
@@ -0,0 +1,141 @@
+aws_api_gateway_rest_api.this: Creation complete after 1s [id=thq85cqumh]
+aws_api_gateway_resource.customers: Creating...
+aws_kms_key.this: Creation complete after 1s [id=bce4f651-696e-4121-b059-7bdfd51e8d04]
+aws_ssm_parameter.db_name: Creation complete after 1s [id=/devops_serverless01/DB_NAME]
+aws_dynamodb_table.this: Creating...
+aws_api_gateway_resource.customers: Creation complete after 1s [id=nzknhh]
+aws_api_gateway_method.customers: Creating...
+aws_api_gateway_method.customers: Creation complete after 0s [id=agm-thq85cqumh-nzknhh-ANY]
+aws_ssm_parameter.api_key: Creation complete after 1s [id=/devops_serverless01/API_KEY]
+aws_iam_role.lambda: Creation complete after 3s [id=devops_serverless01_lambda_role]
+aws_iam_role.apigw: Creation complete after 3s [id=devops_serverless01_apigw_lambda_role]
+aws_api_gateway_account.default: Creating...
+aws_iam_policy.apigw: Creation complete after 4s [id=arn:aws:iam::097922957316:policy/devops_serverless01_apigw_lambda_policy]
+aws_iam_role_policy_attachment.apigw: Creating...
+aws_iam_role_policy_attachment.apigw: Creation complete after 2s [id=devops_serverless01_apigw_lambda_role-20200901100357099900000001]
+aws_dynamodb_table.this: Creation complete after 8s [id=DA_Serverless]
+aws_iam_policy.lambda: Creating...
+aws_api_gateway_account.default: Still creating... [10s elapsed]
+aws_iam_policy.lambda: Creation complete after 5s [id=arn:aws:iam::097922957316:policy/devops_serverless01_lambda_policy]
+aws_iam_role_policy_attachment.lambda: Creating...
+aws_iam_role_policy_attachment.lambda: Creation complete after 2s [id=devops_serverless01_lambda_role-20200901100406867600000002]
+aws_lambda_function.this: Creating...
+aws_lambda_function.this: Creation complete after 6s [id=func_customers]
+aws_cloudwatch_log_group.lambda: Creating...
+aws_api_gateway_integration.lambda: Creating...
+aws_lambda_permission.apigw: Creating...
+aws_lambda_permission.apigw: Creation complete after 1s [id=AllowAPIGatewayInvoke]
+aws_api_gateway_account.default: Still creating... [20s elapsed]
+aws_api_gateway_integration.lambda: Creation complete after 1s [id=agi-thq85cqumh-nzknhh-ANY]
+aws_api_gateway_deployment.prod: Creating...
+aws_cloudwatch_log_group.lambda: Creation complete after 1s [id=/aws/lambda/func_customers]
+aws_api_gateway_deployment.prod: Creation complete after 0s [id=do3d50]
+aws_api_gateway_usage_plan.standard: Creating...
+aws_api_gateway_method_settings.default: Creating...
+aws_api_gateway_method_settings.default: Creation complete after 1s [id=thq85cqumh-prod-*/*]
+aws_api_gateway_usage_plan.standard: Creation complete after 1s [id=bb9tom]
+aws_api_gateway_usage_plan_key.main: Creating...
+aws_api_gateway_usage_plan_key.main: Creation complete after 1s [id=n1apjpnd4e]
+aws_api_gateway_account.default: Creation complete after 27s [id=api-gateway-account]
+
+Apply complete! Resources: 23 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+api_key = vxXt4Ut1Xt4E2hesGZ0Zn6CawannNh6z67shWlxl
```

Comment:
You may remove this output as you have a SSM parameter Secure String. If you want to control access to the parameter (SSM parameter store does not have resource-based policies) you can encrypt it with a KMS CMK and set the permissions in the Key policy.

Reply:
I added the api_key here so I can retrieve it for the `curl` to call the api to create customer.